apps/research-service/src/site-polygons/site-polygon-query.builder.ts

```diff
@@ -123,6 +123,21 @@ export class SitePolygonQueryBuilder extends PaginatedQueryBuilder<SitePolygon>
     return this.where({ projectId: { [Op.in]: filterProjects.map(({ id }) => id) } }, this.siteJoin);
   }
 
+  async filterValidationStatus(validationStatuses: string[]) {
+    const hasNotChecked = validationStatuses.includes("not_checked");
+    const otherStatuses = validationStatuses.filter(status => status !== "not_checked");
```

Comment:
You can avoid going through the list twice by comparing the length of the two lists:
```
    const otherStatuses = validationStatuses.filter(status => status !== "not_checked");
    const hasNotChecked = otherStatuses.length !== validationStatuses.length;
```